org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/panel/InfoPanel.java

```diff
@@ -1385,10 +1411,68 @@ else if (!(last == '*' || last == '-' || last == '+' || last == '/' || last == '
 				}
 			}
 		}
+		return colsql;
+	}
+	
+	/**
+	 * Get SQL FROM clause for ORDER BY
+	 * @param orderColumnInfo
+	 * @param displayColumn
+	 * @return String FROM clause
+	 */
+	private String getFromForOrderBy(ColumnInfo orderColumnInfo, String displayColumn) {
+		String fromClause = "";
+		MTable table = getTable(orderColumnInfo.getAD_Reference_Value_ID(), orderColumnInfo.getColumnName());
+		String tableName = table.getTableName();
+		if(table != null)
+			fromClause += tableName;
+
+		// join translation table
+		if(displayColumn.contains(table.getTableName()+"_Trl")) {
+			String tableNameTrl = tableName+"_Trl";
+			MTable tableTrl = MTable.get(Env.getCtx(), tableNameTrl);
+			String sqlSelect = orderColumnInfo.getSelectClause();
+			String[] keyCols = tableTrl.getKeyColumns();
+
+			fromClause += " JOIN " + tableNameTrl + " ON (";
+			for(int i = 0; i < keyCols.length; i++) {
+				String keyCol = keyCols[i];
+				
+				if(i > 0)
+					fromClause += " AND ";
+				
+				fromClause += tableNameTrl + "." + keyCol + " = ";
+				
+				if("AD_Language".equalsIgnoreCase(keyCol))
+					fromClause += " '" + Env.getAD_Language(Env.getCtx()) + "' ";
+				else
+					fromClause += sqlSelect;
+			}
+			fromClause += ") ";
+		}
+		return fromClause;
+	}
+	
+	/**
+	 * Get WHERE clause for ORDER BY
+	 * @param orderColumnInfo
+	 * @return String WHERE clause
+	 */
+	private String getWhereForOrderBy(ColumnInfo orderColumnInfo) {
+		MTable table = getTable(orderColumnInfo.getAD_Reference_Value_ID(), orderColumnInfo.getColumnName());
+		String tableName = table.getTableName();
+		String[] keyCols = table.getKeyColumns();
+		String sqlSelect = orderColumnInfo.getSelectClause();
+		String whereClause = "";
 		
-		return String.format(" ORDER BY %s %s ", colsql, isColumnSortAscending? "" : "DESC");
+		for(String keyCol : keyCols) {
+			if(!Util.isEmpty(whereClause))
+				whereClause += " AND ";
+			whereClause += tableName + "." + keyCol + " = " + sqlSelect;
+		}
```

Comment:
This for loop is actually pointless. There's only 1 sqlSelect value, this will only works if keyCols is of length 1.